base_layer/core/src/base_node/tari_pulse_service/mod.rs

```diff
@@ -0,0 +1,216 @@
+use std::{str::FromStr, time::Duration};
+
+use futures::future;
+use hickory_client::{
+    client::{AsyncDnssecClient, ClientHandle},
+    proto::{
+        iocompat::AsyncIoTokioAsStd,
+        rr::dnssec::{public_key::Rsa, SigSigner, TrustAnchor},
+        xfer::DnsMultiplexer,
+    },
+    rr::{DNSClass, Name, RData, Record, RecordType},
+    tcp::TcpClientStream,
+};
+use log::{error, info};
+use serde::{Deserialize, Serialize};
+use tari_p2p::Network;
+use tari_service_framework::{async_trait, ServiceInitializationError, ServiceInitializer, ServiceInitializerContext};
+use tari_shutdown::ShutdownSignal;
+use tari_utilities::hex::Hex;
+use tokio::{net::TcpStream as TokioTcpStream, sync::watch, time};
+
+use super::LocalNodeCommsInterface;
+use crate::base_node::comms_interface::CommsInterfaceError;
+
+const LOG_TARGET: &str = "c::bn::tari_pulse";
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct TariPulseConfig {
+    pub check_interval: Duration,
+    pub network: Network,
+}
+
+impl Default for TariPulseConfig {
+    fn default() -> Self {
+        Self {
+            check_interval: Duration::from_secs(120),
+            network: Network::default(),
+        }
+    }
+}
+
+fn get_network_dns_name(network: &Network) -> Name {
+    match network {
+        Network::NextNet => Name::from_str("checkpoints-nextnet.tari.com").unwrap(),
+        _ => panic!("Network not supported"),
```

Comment:
have you tested this running any network except nextnet?
This panic should not be here. Properly map it out. Even thou the other networks are not supported yet, add their DNS, they can fail connectivity if need be.